src/components/main/fruits/fruitsListSection/__test__/FruitListContainer.test.tsx

```diff
@@ -1,137 +1,244 @@
             )
           ).not.toBeInTheDocument();
-          expect(errorMessage).toBeInTheDocument();
+        });
+
+        describe("When unsuccesfull finding data", () => {
+          beforeEach(() => {
+            jest
+              .spyOn(myService, "getDataFromApi")
+              .mockResolvedValue(mockFruitListFromApi);
+          });
+
+          it("displays an error", async () => {
+            render(
+              <HashRouter>
+                <FruitListContainer />
+              </HashRouter>
+            );
+
+            const userSearch = "test";
+            const input = screen.getByTestId("input");
+            const errorMessage = await screen.findByTestId("error-message");
+            let foundFruit;
+
+            // eslint-disable-next-line testing-library/no-unnecessary-act
+            act(() => {
+              userEvent.type(input, userSearch); //.type simulates user typing--needs 2 arguments (an element and a text)
+            });
+            await waitFor(() => {
+              foundFruit = screen.queryByTestId(userSearch);
+            });
+
+            //screen.debug();
+            expect(foundFruit).not.toBeInTheDocument();
+            expect(
+              screen.queryByTestId(
+                mockFruitListFromApi[0].name.toLocaleLowerCase()
+              )
+            ).not.toBeInTheDocument();
+            expect(errorMessage).toBeInTheDocument();
+          });
         });
       });
     });
   });
 });
 
-////////////FROM FRUIT LIST COMPONENT ////////////////
-
-describe("FruitList section calls API services", () => {
-  describe("when succesfull", () => {
-    beforeEach(() => {
-      jest
-        .spyOn(myService, "getDataFromApi")
-        .mockResolvedValue(mockFruitListFromApi);
-    });
-
-    it("renders FruitList section", async () => {
-      render(
-        <HashRouter>
-          <FruitListContainer />
-        </HashRouter>
-      );
+describe("When there is no data", () => {
+  beforeEach(() => {
+    jest.spyOn(myService, "getDataFromApi").mockResolvedValue([]);
+  });
 
-      const fruit0 = mockFruitListFromApi[0].name.toLowerCase();
-      const fruit1 = mockFruitListFromApi[1].name.toLowerCase();
+  it("displays an error", async () => {
+    render(<FruitListContainer />);
 
-      // eslint-disable-next-line testing-library/no-unnecessary-act
-      await act(async () => {
-        screen.queryByTestId(fruit0);
-      });
+    const errorMessage = await screen.findByTestId("error-message");
 
-      expect(screen.getByTestId(fruit0)).toBeInTheDocument();
-      expect(screen.getByTestId(fruit1)).toBeInTheDocument();
-    });
+    expect(errorMessage).toBeInTheDocument();
   });
+});
 
-  describe("when not succesfull", () => {
-    beforeEach(() => {
-      jest.spyOn(myService, "getDataFromApi").mockResolvedValue([]);
-    });
-
-    it("displays an error", async () => {
-      render(<FruitListContainer />);
+////////////////////////////////////////////////////
```

Comment:
i know you mentioned something regarding the commented code, but do we really needed to merge it? don't think so. Let's get rid of it :)